### DISCUSSION TITLE
How to update <x-select> when another select has changed

### DISCUSSION THREAD

**Discussion by rabol** [wireui/wireui#636] [Q&A]
Hi

I'm trying to create a component that have 3 <x-select> Country, State and City

I have 3 Api routes, one for Country, State and City

How do I get the 3 <x- select> to talk together in a livewire component?

**Comment by rhonautix** [comment]
Have you figured it out? 

I'm trying to accomplish the same. I managed to make the first `<x-select>` to update the options on the second `<x-select>`. The problem is that if I change the selected item on first `<x-select>`, the selected item on the second `<x-select>` does not reset even though the options have already changed.

**Reply by rabol** [reply]
No I have not solved it yet
Care to share what you have - maybe another set of eyes can help :)

**Reply by rhonautix** [reply]
```
public function mount(){
        $this->companies = Employer::pluck('company_name', 'id');
        $this->billings = collect();  //initial empty options
     }


public function updatedCompanyId($id)
    {
        $this->company_id = $id;
        $company = Employer::find($id);
        
        //populate options on change
        $this->billings = $company ? $company->billings->pluck('invoice_id','id') : collect();
    }
```
```
<div class="col-span-1">
                <div class="mb-2">
                    <x-select label="Company Name:" placeholder="Select a company"
                              :options="$this->companies"
                              :option-key-value=true
                              wire:model="company_id" />
                </div>
            </div>
            <div class="col-span-1">
                <div class="mb-2">
                    <x-select label="Billing Number:" placeholder="Select a billing"
                              :options="$this->billings"
                              :option-key-value=true
                              wire:model="billing_id" />
                </div>
            </div>
```

Selected company has billing, first option on the billing is selected:

![image](https://user-images.githubusercontent.com/115517521/230137003-db1ac4c9-5267-47b2-9d9e-4343dbe12a65.png)

Selected company has **changed** but the billing `<x-select>` did not reset and retains the previously selected option.

![image](https://user-images.githubusercontent.com/115517521/230138463-4f47c09d-cfde-43d3-8c5b-141509ca2f4e.png)

Is there a method or event that I can trigger to reset?

**Reply by rhonautix** [reply]
I figured it out with some help from ChatGPT ;)

```
    public function updatedCompanyId($id)
    {
        $this->reset('billing_id');
        $this->resetValidation('billing_id');

        $this->company_id = $id;
        $company = Employer::find($id);
        $this->billings = $company ? $company->billings->pluck('invoice_id','id') : collect();

    }
```

**Reply by rabol** [reply]
Thanks for the input - I'll see If I can make mine work

**Comment by Uhasith** [comment]
I'm trying to set a value automatically when i select something on first select input. I called `x-on:selected="@this.setConsumer($event.detail.value)` when user select value for first input. then i set a new value to my second select filed inside that method. Value is changing but UI not updating. I do not have find a solution for this yet. 

```
    public function setConsumer($id){

        $this->reset('consumer_id');

        $office = Office::find($id);
        $consumer = $office->consumer->first();

        if(empty($consumer)){

            $this->consumer_id = null;
            return;
         }

        $this->consumer_id  =  $consumer->id;
    }
    ```
    
    @rabol @rhonautix
